### ISSUE TITLE
Call 2 table with same name in 2 different 4D Server in one project

### ISSUE THREAD

**Issue by DomTkt** [Wakanda/wakanda-issues#173]
Hello, i would like to know if someone succed to call 2 table with same name in one project.
Thanks.

**Comment by fredGandelot** [comment]
Hi @DomTkt  from server side to client side, it's not possible to difference 2  tables with the same name. 

From **`wakadanda2.0`**, on the server side it can be  however bypassed  by adding your remote models instead of merging them . (on your solution , right click on the models forlder and click new model > select 4D , and uncheck the box `merge with active model`

![capture d ecran 2017-04-11 a 14 05 06](https://cloud.githubusercontent.com/assets/4920957/24908328/e933b368-1ebf-11e7-97cd-c76a8bef2493.png)


This way you won't access your model via the `ds` object but with its  `DataStoreName` (in my exemple Untitled1 

ex=
```
untitled1.dataclass1.all() ;
```
Thus you won"t  have any collusion between your 2 models.

**Comment by fredGandelot** [comment]
@DomTkt  anyhow in your case I would suggest to rename your table name.

**Comment by DomTkt** [comment]
untitled1.dataclass1.all() ;
does not work i try in angular : 

ds.Untitled1.$query({orderBy:"ID desc",pageSize:3}).$promise.then(function(response) {
         $scope.favoriteUntitled1 = response.result; 
       });

**Comment by fredGandelot** [comment]
@DomTkt  
Are using wakV2 ? 
- Did you add the model as previously explained ? (uncheck the box in the dialog)
- Then test it on the backend first - If it's ok :
- On the front side instantiate your added model 

`var addedModel = new WakandaClient.WakandaClient({ catalog : "My4DModel"});`
then : 
```
 addedModel.getCatalog(['']).then(function(My4DModel){
          console.log(My4DModel);
          My4DModel.Company4.query().then(function(result2)
      {
    console.log(result2);
      })
}

```

**Comment by DomTkt** [comment]
@fredGandelot Thanks for your answer !

Yes i use wakV2, i uncheck merge with active model

i'm going to test firstly in backend

**Comment by DomTkt** [comment]
@fredGandelot I have in back end when i try Base1.MyTable.all() : 

{
	"__ERROR": [
		{
			"message": "Base1 is not defined, line #1, file:///C:/Users/PHAN%20Dominique/Documents/Wakanda/solutions/TestTest/backend/script/addAnom.js.",
			"componentSignature": "jvsc",
			"errCode": 4055
		},
		{
			"message": "Error evaluating javascript file:///C:/Users/PHAN Dominique/Documents/Wakanda/solutions/TestTest/backend/script/addAnom.js",
			"componentSignature": "JS4D",
			"errCode": 2
		}
	]
}

**Comment by DomTkt** [comment]
I try : 
     var addedModel = new WakandaClient.WakandaClient({ catalog : "Base1"});
    	
    addedModel.getCatalog(['']).then(function(Base1){
          console.log(Base1);
          Base1.Depots.query().then(function(result2)          
      {
    console.log(result2);
      })
});

but return : 

Uncaught (in promise) TypeError: Cannot read property 'query' of undefined
    at home.js:12

**Comment by fredGandelot** [comment]
@DomTkt  

1- When the connection is performed Wakanda studio should display the remote added Model. if it's not the case , check the connection properties.
2 - Once the remote schema is displayed in the studio, launch the server
3-  You must use the **Wakanda Enterprise  Server version with a license** attached to it.
4 - if your license.key is not valid a dialog will pop up 
5 - If no pop up occurs,  it should work (you can easy check the connection via rest API 
http://localhost/rest/Base1/$catalog	  //it displays only the dataclasses from the added 4DModel model ) be sure to use the correct `port` (8081 by default)

**Comment by fredGandelot** [comment]
@DomTkt  do not try to access your data on the front side  if you don"t even access them on the server or via the REST API :) (see my last comment)

**Comment by DomTkt** [comment]
1 - The model is permormed : 

![image](https://cloud.githubusercontent.com/assets/12907650/24997890/90aaf182-2038-11e7-8d4c-9af47613d480.png)

2- The Server is connected successfull : 

![image](https://cloud.githubusercontent.com/assets/12907650/24997983/df282b40-2038-11e7-92a1-b15131182159.png)

5- no pop up occurs and when i try http://localhost:8081/rest/Base1/$catalog => return : 

{"__ERROR":[{"message":"Error code: 1800\nThe \"Base1\" datastore class cannot be found\ncomponent: 'dbmg'\ntask 10160, name: 'HTTP_WORKER_THREAD'\n","componentSignature":"dbmg","errCode":1800}]}

**Comment by fredGandelot** [comment]
@DomTkt  > Do you use Wakanda Enterprise Version ? Do u have a license.key ?

**Comment by DomTkt** [comment]
Yep i use Wakanda Enterprise Version, i don't know if i have license.key but in version : 
Wakanda Enterprise Studio  11 build 11.197492 
all work

**Comment by DomTkt** [comment]
@fredGandelot @hamzahik

**Comment by fredGandelot** [comment]
@DomTkt  Yes it did work in wak build 11.197492, but `wakanda v2.0` introduces the license.key control.

Please visite wakanda.github.io , and download Wakanda Enterprise Version (trial version for example) > Fill the fields > You will get an email > Click to confirm (add me to the list) 
THen you'll receive another email with a link to download the license.key :)

Regards fred

**Comment by DomTkt** [comment]
@fredGandelot Yes i do this but it's the same result :/

**Comment by fredGandelot** [comment]
@DomTkt  where did you store your license.key file ? It should be in \documents\wakanda\license\license.key

**Comment by DomTkt** [comment]
Yes i put license.key in /documents/wakanda, i don't have folder license so i create it and i put the file,

To be sure I put the file everywhere

But i don't think it's the probleme because 
I no longer have the window with the license problem

**Comment by fredGandelot** [comment]
I to guarantee you that it could'nt have worked without a license. So it's not maybe the only problem it was one ^^ .  

http://localhost:8081/rest/$catalog => what do u get ?
http://localhost:8081/rest/Base1/$catalog  => what do u get ?

How many models do u have in your solution  ? only one added ? 
Have you tried to reload the model once the connection is established ? 

otherwise send me your solution, I have a look at it .

note : yes you have to create the folder license if it doesn"t exist

>>

**Comment by DomTkt** [comment]
@fredGandelot I try to create an other project and i check merge with the model and the rest/dataclass work ! but also i can't access 2 dataclass(same name) with 2 database

**Comment by DomTkt** [comment]
@fredGandelot I restart my computer and it's work , i don't undestand haha

**Comment by fredGandelot** [comment]
Well glad it worked 👍

**Comment by fredGandelot** [comment]
@DomTkt  can u close this issue ?

**Comment by DomTkt** [comment]
@fredGandelot wait because i can't access when i do 
 	var addedModel = new WakandaClient.WakandaClient({ catalog : "Untitled1"});
 
  	
 addedModel.getCatalog(['']).then(function(Untitled1){
  		
  		Untitled1.SameName.$query({orderBy:"ID desc",pageSize:3}).$promise.then(function(response) {
         $scope.tasks = response.result; 
       });
      

but when i do : 

 $wakandaManager.getDataStore().then(function(ds) {
      console.log('Angular-Wakanda is ready!');
      ds.Clients.$query({orderBy:"ID desc",pageSize:3}).$promise.then(function(response) {
         $scope.favoriteSuperheroes = response.result; 
       });

its work

**Comment by DomTkt** [comment]
@hamzahik  As-tu une idée de comment je peux accéder à 2 tables identiques sur 2 bases différentes sur Wakanda 2.0 ?

J'ai essayé la méthode de @fredGandelot avec : 

var addedModel = new WakandaClient.WakandaClient({ catalog : "Untitled1"});
addedModel.getCatalog(['']).then(function(Untitled1){
    Untitled1.SameName.$query({orderBy:"ID desc",pageSize:3}).$promise.then(function(response) {
    $scope.tasks = response.result; 
});

il ne m'affiche rien sur ma page néanmoins sa méthode est reconnu mais je ne suis pas sûr que ça soit la bonne.
